frontend/src/features/manageProfile/apiManager/manageProfileAPI.tsx

```diff
@@ -123,19 +123,9 @@ export const getUserRolesByCompanyId = (): Promise<string[]> => {
  * Retrieves the roles of an IDIR user (i.e., OnRouteBC staff).
  */
 export const getIDIRUserRoles = async (): Promise<RequiredOrNull<string[]>> => {
-  try {
-    const response = await httpGETRequest(`${VEHICLES_URL}/users/roles`);
-    if (response.status === 200) {
-      return response.data as string[];
-    }
-
-    // Error status code, return null for roles
-    return null;
-  } catch (err) {
-    // If error, return null for roles
-    console.error(err);
-    return null;
-  }
+  return httpGETRequest(`${VEHICLES_URL}/users/roles`).then(
+    (response) => response.data,
+  );
 };
```

Comment:
Is there a reason why we should do this over using the try-catch block statement (as well as async/await)?

Reply:
It masks the actual error which does not trigger 1) retry 2) register as an error with react query.
This leads to an incomplete state update causing a perennially running loading screen.
<img width="784" alt="image" src="https://github.com/bcgov/onroutebc/assets/84348052/4ed413b1-e466-4d34-a1ec-b482dbc81e55">

Reply:
ok